src/lib/builtins/symbol.rs

```diff
@@ -0,0 +1,65 @@
+use crate::{
+    builtins::{
+        object::{Object, PROTOTYPE},
+        value::{to_value, ResultValue, Value, ValueData},
+    },
+    exec::Interpreter,
+};
+use gc::{Gc, GcCell};
+use rand::random;
+
+/// https://tc39.es/ecma262/#sec-symbol-description
+/// Creates Symbol instances.
+///
+/// Symbol instances are ordinary objects that inherit properties from the Symbol prototype object.
+/// Symbol instances have a [[SymbolData]] internal slot.
+/// The [[SymbolData]] internal slot is the Symbol value represented by this Symbol object.
+pub fn call_symbol(_: &Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
+    // From an implementation and specificaition perspective Symbols are similar to Objects.
+    // They have internal slots to hold the SymbolData and Description, they also have methods and a prototype.
+    // So we start by creating an Object
+    // TODO: Set prototype to Symbol.prototype (by changing to Object::create(), use interpreter to get Symbol.prototype)
+    let mut sym_instance = Object::default();
+
+    // Set description which should either be undefined or a string
+    let desc_string = match args.get(0) {
+        Some(value) => to_value(value.to_string()),
+        None => Gc::new(ValueData::Undefined),
+    };
+
+    sym_instance.set_internal_slot("Description", desc_string);
+    sym_instance.set_internal_slot("SymbolData", to_value(random::<i32>()));
```

Comment:
What is this slot used for?

Reply:
All symbols need to be unique and have some sort of ID.
Symbols can be used as field names in objects so the SymbolData number would be used there in sym_properties